handlers/common.go

```diff
@@ -130,27 +128,214 @@ func GetValidatorEarnings(validators []uint64, currency string) (*types.Validato
+	if elApr31d < float64(-1) {
+		elApr31d = float64(-1)
+	}
+	clApr365d := (float64(income.ClIncome365d) / float64(totalDeposits))
+	if clApr365d < float64(-1) {
+		clApr365d = float64(-1)
+	}
+
+	elApr365d := (float64(income.ElIncome365d) / float64(totalDeposits))
+	if elApr365d < float64(-1) {
+		elApr365d = float64(-1)
 	}
 
 	return &types.ValidatorEarnings{
-		Total:                c.EarningsTotal,
-		LastDay:              c.EarningsLastDay,
-		LastWeek:             c.EarningsLastWeek,
-		LastMonth:            c.EarningsLastMonth,
-		APR:                  apr,
-		TotalDeposits:        totalDeposits,
-		TotalWithdrawals:     totalWithdrawals,
-		LastDayFormatted:     utils.FormatIncome(c.EarningsLastDay, currency),
-		LastWeekFormatted:    utils.FormatIncome(c.EarningsLastWeek, currency),
-		LastMonthFormatted:   utils.FormatIncome(c.EarningsLastMonth, currency),
-		TotalFormatted:       utils.FormatIncome(c.EarningsTotal, currency),
-		TotalChangeFormatted: utils.FormatIncome(c.EarningsTotal+totalDeposits, currency),
+		Income1d: types.ClElInt64{
+			El:    income.ElIncome1d,
+			Cl:    income.ClIncome1d,
+			Total: earnings1d,
+		},
+		Income7d: types.ClElInt64{
+			El:    income.ElIncome7d,
+			Cl:    income.ClIncome7d,
+			Total: earnings7d,
+		},
+		Income31d: types.ClElInt64{
+			El:    income.ElIncome31d,
+			Cl:    income.ClIncome31d,
+			Total: earnings31d,
+		},
+		Apr7d: types.ClElFloat64{
+			El:    elApr7d,
+			Cl:    clApr7d,
+			Total: clApr7d + elApr7d,
+		},
+		Apr31d: types.ClElFloat64{
+			El:    elApr31d,
+			Cl:    clApr31d,
+			Total: clApr31d + elApr31d,
+		},
+		Apr365d: types.ClElFloat64{
+			El:    elApr365d,
+			Cl:    clApr365d,
+			Total: clApr365d + elApr365d,
+		},
+		ElIncomeTotal:        income.ElIncomeTotal,
+		TotalDeposits:        int64(totalDeposits),
+		LastDayFormatted:     utils.FormatIncome(earnings1d, currency),
+		LastWeekFormatted:    utils.FormatIncome(earnings7d, currency),
+		LastMonthFormatted:   utils.FormatIncome(earnings31d, currency),
+		TotalFormatted:       utils.FormatIncome(income.ClIncomeTotal, currency),
+		TotalChangeFormatted: utils.FormatIncome(income.ClIncomeTotal+int64(totalDeposits), currency),
 	}, balancesMap, nil
 }
 
+// getProposalLuck calculates the luck of a given set of proposed blocks for a certain number of validators
+// given the blocks proposed by the validators and the number of validators
+//
+// precondition: slots is sorted by ascending block number
+func getProposalLuck(slots []uint64, validatorsCount int) float64 {
+	// Return 0 if there are no proposed blocks or no validators
+	if len(slots) == 0 || validatorsCount == 0 {
+		return 0
+	}
+	// Timeframe constants
+	fiveDays := time.Hour * 24 * 5
+	oneWeek := time.Hour * 24 * 7
+	oneMonth := time.Hour * 24 * 30
+	sixWeeks := time.Hour * 24 * 45
+	twoMonths := time.Hour * 24 * 60
+	threeMonths := time.Hour * 24 * 90
+	fourMonths := time.Hour * 24 * 120
+	fiveMonths := time.Hour * 24 * 150
```

Comment:
Could use `utils.Day` and `utils.Month`.

Reply:
adressed in new commit, ptal